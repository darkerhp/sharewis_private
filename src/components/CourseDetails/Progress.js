import React from 'react';
import ReactNative from 'react-native';

const { PropTypes } = React;
const { View, StyleSheet, Text } = ReactNative;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'space-around',
  },
  barContainer: {
    flexDirection: 'row',
    height: 18,
    borderWidth: 1,
    borderColor: '#cdcdcd',
  },
  complete: {
    backgroundColor: '#427fda',
  },
  incomplete: {
    backgroundColor: '#cdcdcd',
  },
});

const t = {
  lectureCompleteLable: 'レクチャーが完了しました',
};

const Progress = ({ completeLectureCount, totalLectureCount }) => {
  const completePercentage = completeLectureCount > 0
    ? (completeLectureCount / totalLectureCount) * 100 : 0;
  const incompletePercentage = 100 - completePercentage;
  return (
    <View style={styles.container}>
      <View style={styles.progressText}>
        <Text>{completeLectureCount}/{totalLectureCount}{t.lectureCompleteLable}</Text>
      </View>
      <View style={styles.barContainer}>
        <View style={[styles.complete, { flex: completePercentage }]} />
        <View style={[styles.incomplete, { flex: incompletePercentage }]} />
      </View>
    </View>
  );
};

Progress.propTypes = {
  completeLectureCount: PropTypes.number.isRequired,
  totalLectureCount: PropTypes.number.isRequired,
};

export default Progress;
