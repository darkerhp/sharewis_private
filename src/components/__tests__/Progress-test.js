/* eslint-disable no-undef */
/* eslint-disable import/no-extraneous-dependencies */
import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import Progress from '../CourseDetails/Progress';

jest.mock('react-native-i18n', () => ({ t: () => '' }));
jest.mock('../ProgressBar', () => 'ProgressBar');

describe('Progress', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Progress
        completeLectureCount={0}
        totalLectureCount={10}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
