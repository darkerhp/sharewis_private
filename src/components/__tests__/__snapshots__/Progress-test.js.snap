exports[`Progress renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#4a4a4a",
        "marginBottom": 5,
      }
    }>
    0/10 [missing \"en.progressText\" translation]
  </Text>
  <ProgressBar
    progress={0} />
</View>
`;
